primitives/src/adapter.rs

```diff
@@ -33,14 +33,10 @@ impl fmt::Display for AdapterError {
     }
 }
 
-#[derive(Debug, Clone)]
-pub enum AdapterOptions {
```

Comment:
We actauly don't need this, see the `main.rs` of the `validator_worker` and `sentry`. We instantiate them separately, this is not part of the Behaviour of the struct (i.e. behaviour is what Traits should represent, like interfaces in classical OOP).